File: Sorting and Searching/Stick Lengths/stickLengths.java
import java.util.Arrays;
import java.util.Scanner;

public class stickLengths {
    private static void solve(){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        long ans = 0;
        int[] arr = new int[n];
        for(int i=0;i<n;i++){
            arr[i] = scanner.nextInt();
        }
        Arrays.sort(arr);
        int x = arr[n/2];
        for(int i=0;i<n;i++){
            ans += Math.abs(x-arr[i]);
        }
        System.out.println(ans);
        return;

    }
    public static void main(String args[]){
        solve();
    }
}

File: Sorting and Searching/Stick Lengths/stick_lengths.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int


void solve(){
	ll n, ans = 0, temp;
	cin>>n;
	vector<int> arr(n);

	for(int i=0;i<n;i++){
		cin>>temp;
		arr[i] = temp;
	}
	sort(arr.begin(), arr.end());
	int x = arr[n/2];
	for(int i=0;i<n;i++){
		ans += abs(x-arr[i]);
	}
	cout<<ans;
    return;
	
}

int main(){
	long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	return 0;
}